dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -289,14 +342,20 @@ void WebdatasetLoader::PrepareMetadataImpl() {
   output_indicies_.reserve(ext_.size());
 
   std::vector<size_t> dtype_sizes_(dtypes_.size());
-  for (size_t i = 0; i < dtypes_.size(); i++)
+  for (size_t i = 0; i < dtypes_.size(); i++) {
     dtype_sizes_[i] = TypeTable::GetTypeInfo(dtypes_[i]).size();
+  }
 
-  for (size_t wds_shard_index = 0; wds_shard_index < index_paths_.size(); wds_shard_index++) {
+  for (size_t wds_shard_index = 0; wds_shard_index < paths_.size(); wds_shard_index++) {
     unfiltered_samples.resize(0);
     unfiltered_components.resize(0);
-    detail::wds::ParseIndexFile(unfiltered_samples, unfiltered_components,
-                                index_paths_[wds_shard_index]);
+    if (is_index_generated_) {
+      detail::wds::ParseTarFile(unfiltered_samples, unfiltered_components,
+                                wds_shards_[wds_shard_index]);
```

Comment:
Maybe DALI can print something that there is no index files provided and we are parsing the tar itself.

Reply:
alright